docs/DeveloperGuide.md

```diff
@@ -1,34 +1,384 @@
+
+
+## **Getting Started**
+Refer to the setting up guide over [here](https://github.com/AY2021S1-CS2113T-W11-1/tp/releases).
+
+
+
+
+
+
+## **Design**
+
+### Architecture
+
+The figure below shows a high-level design for the architecture of Zoomaster. 
+![](https://github.com/TYS0n1/tp/blob/team-Branch2/docs/diagrams/archiveture2-01.png?raw=true)
+*<center/> Figure 1.1 Architecture diagram of Zoomaster </center> <br/></br>*
+
+Our Program can be split up into 8 components
+* Initialization
+* User Interface
+* Parser
+* Commands
+* Temp Lists
+* Storage
+* Local Files
+* Exceptions
+
+These components interact with each other as shown in Figure 1.1 to execute the functionalities of Zoomaster.
+
+### Initialization
+The diagram below shows a class-level diagram for Zoomaster. <br/></br>
+![](https://raw.githubusercontent.com/Speedweener/ip/master/docs/images/initial.png)
+*<center/> Figure 1.2 Class diagram of Initialization </center> <br/></br>*
+
+**API**:`Zoomaster.java`
+
+The Initialization component is responsive for setting up Zoomaster for it to be used by users. 
+
+It consists of `Zoomaster`, `Ui`, `Storage`, `BookmarkList`, `Timetable` and `Module` classes.
+
+Its main roles are:
+* Retrieving bookmark, timetable and planner lists from storage if it exists.
+* Set the list of Modules for Zoomaster 
+* Initializes the User Interface object, as well as the 3 Storage  objects for `BookmarkList`, `Timetable` and `Planner`.
+
+### User Interface component
+
+![](images/UiClassDiagram.png)
+*<center/> Figure 1.3 Class diagram of User Interface </center> <br/></br>*
+
+**API**:`Ui.java`
+
+The UI component is responsible for all visual output to the User from the app. 
+
+The only class carrying out the component's function is the `Ui` class.
+
+It utilises the `Jansi` library to enable the usage of ANSI escape codes to format the console, allowing the app to output in colour. <br/>
+
+The UI also receives input from the User using a `Scanner` object. It returns the input as a String to the main function. <br/>
+
+In addition, the UI contains the different exception and error messages which can be displayed. When a particular exception is thrown (eg. **UNKNOWN_INPUT**), the corresponding function is called in UI to print out the error message (**printUnknownInputMessage()**). 
+
+
+Its main roles are:
+
+* Receiving user commands and returning it to the Main function.
+* Prints visual output in the console for the User
+
+
+### Parser component
+
+
+
+![]()
+*<center/> Figure 1.4 Class diagram of Parser </center> <br/></br>*
+
+
+The Parser component is responsible for decoding the user's input and telling the Main function which command to execute.
+
+It also contains the **programMode** which indicates which mode the program currently is in.
+
+
+It initializes the different commands according to which mode the program currently is in.
+
+It consists of `Parser` and the Command interface classes.
```

Comment:
Maybe use point form for these information? Seems like you guys were trying to list things out